src/validations.test.ts

```diff
@@ -0,0 +1,133 @@
+import assert from "node:assert";
+import test, { describe } from "node:test";
+import { PaymentOptions } from "./types.js";
+import {
+	validateBankAccount,
+	validateDataModel,
+	validateSimplePayment,
+	ValidationError,
+	ValidationErrorMessage,
+} from "./validations.js";
+
+const iban = "LC14BOSL123456789012345678901234";
+const validBankAccount = {
+	iban,
+};
+
+describe("validateBankAccount", () => {
+	const path = "payments[0].bankAccounts[0]";
+	test("validate IBAN", () => {
+		assert.throws(
+			() =>
+				validateBankAccount({
+					iban: "1234567890",
+				}, path),
+			new ValidationError(ValidationErrorMessage.InvalidIBAN, `${path}.iban`),
+		);
+		assert.doesNotThrow(() => validateBankAccount(validBankAccount, path));
+	});
+
+	test("validate BIC", () => {
+		assert.throws(
+			() =>
+				validateBankAccount({
+					iban,
+					bic: "123",
+				}, path),
+			new ValidationError(ValidationErrorMessage.InvalidBIC, `${path}.bic`),
+		);
+		assert.throws(
+			() =>
+				validateBankAccount({
+					iban,
+					// todo: question is empty string valid input for arbitrary fields?
```

Comment:
I'm not sure. It depends on how different applications handle serialization. Some might accept it, others might not. For now, let's treat empty as valid.

Reply:
[Don't throw, if arbitrary fields are equal to empty string.](https://github.com/xseman/bysquare/pull/43/commits/57abaebb217311017ed91fee9148a5df91805559)